### PR TITLE
NMR-5272: add check whether NMRData is FID or not and added new alert…

### PR DESCRIPTION
**Pull request by JannalieT**
… message if trying to load an FID as a dataset, remove any NMRViewDatasets that have been added to the project during incorrect loading

### CODE REVIEW

File: nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/bruker/BrukerData.java [hunk 1]
```diff
@@ -1818,6 +1818,11 @@ public boolean isFrequencyDim(int iDim) {
         return result;
     }
 
+    @Override
+    public boolean isFID() {
+        return !isFrequencyDim(0);
+    }
+
```
Comment (JannalieT) [inline nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/bruker/BrukerData.java:1825]: Not sure if this is an okay way to tell whether bruker data is an FID, just following how NMRViewData did it here https://github.com/nanalysis/nmrfx/blob/3b332a81a2608a48ba937cb58e138df493ede4d5/nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/nmrview/NMRViewData.java#L375

**Review by tbparis — COMMENTED** [review]
Not much comment on the code itself.

From what I tested:
- open a Spinit -> export JCamp data-> processing it and saving the nv file
- then try to open a dx file with "open dataset", try to open a nv file with open FID, everything works as expected regarding the ticket, I got the error message.
From what I understand, NmrFx dataset only contains processed data and not FID, and JCamp always contain an FID. The "verify" method is based on that.

So behavior is correct.

I just have a question to be sure to understand. The way "verify" is implemented seems an indirect way to check that "Open FID" open an acquisition data instead of an NmrFx dataset, and to check the reverse too, Why not checking the extension of the file? If it is ".nv" the user did a mistake.
Is there cases where the extension is ambiguous and can be an FID or only processed data?

File: nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/rs2d/RS2DData.java [hunk 2]
```diff
@@ -210,6 +210,17 @@ public static boolean findFID(StringBuilder bpath) {
...
 
+    @Override
+    public boolean isFID() {
+        if (header.isPresent(RS2DParam.STATE)) {
+            List<Integer> states = header.getInts(RS2DParam.STATE);
+            if (states != null) {
+                return states.get(0) == 0;
+            }
+        }
+        return true;
```
[the remark this answers is not in the document]
Comment (JannalieT) [reply nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/rs2d/RS2DData.java:223]: Added a debugging message

**Comment by JannalieT** [comment]
> I just have a question to be sure to understand. The way "verify" is implemented seems an indirect way to check that "Open FID" open an acquisition data instead of an NmrFx dataset, and to check the reverse too, Why not checking the extension of the file? If it is ".nv" the user did a mistake. Is there cases where the extension is ambiguous and can be an FID or only processed data?

For the JCAMP files or the RS2D files (data.dat) the contents could be FID or processed, but the file extension is the same, so I don't think we can rely on the file extension to check

**Review by tbparis — APPROVED** [review]
Ok, I understand better, that's why we infer it from parameter we can read.

I don't have any more question, and no more bug when I try to reproduce it.

**Pull request merged**